src/test/java/uk/ac/cam/cl/dtg/isaac/dao/EventBookingPersistenceManagerTest.java

```diff
@@ -0,0 +1,138 @@
+    expect(mockContentManager.getContentById("event2")).andReturn(eventPage2);
+
+    Connection dummyConnection = createMock(Connection.class);
+    PreparedStatement dummyPreparedStatement = createMock(PreparedStatement.class);
+    ResultSet dummyResultSet = createMock(ResultSet.class);
+
+    expect(mockEventsDatabase.getDatabaseConnection()).andReturn(dummyConnection);
+    expect(dummyConnection.prepareStatement(expectedQuery)).andReturn(dummyPreparedStatement);
+    dummyPreparedStatement.setString(1, "event1");
+    dummyPreparedStatement.setString(2, "event2");
+    expect(dummyPreparedStatement.executeQuery()).andReturn(dummyResultSet);
+
+    expect(dummyResultSet.next()).andReturn(true);
+    prepareEventBookingResultSet(dummyResultSet, 1L, "event1", now);
+
+    expect(dummyResultSet.next()).andReturn(true);
+    prepareEventBookingResultSet(dummyResultSet, 2L, "event2", now);
+
+    expect(dummyResultSet.next()).andReturn(false);
+    dummyResultSet.close();
+    dummyPreparedStatement.close();
+    dummyConnection.close();
+
+    expect(mockAccountManager.getUserDTOById(1L, true)).andReturn(bookedUser1);
+    expect(mockAccountManager.convertToUserSummary(bookedUser1, UserSummaryWithEmailAddressAndGenderDTO.class)).andReturn(bookedUserSummary1);
+
+    expect(mockAccountManager.getUserDTOById(2L, true)).andReturn(bookedUser2);
+    expect(mockAccountManager.convertToUserSummary(bookedUser2, UserSummaryWithEmailAddressAndGenderDTO.class)).andReturn(bookedUserSummary2);
+
+    Object[] mockedObjects = {mockEventsDatabase, mockAccountManager, mockContentManager, mockObjectMapper,
+        dummyConnection, dummyPreparedStatement, dummyResultSet};
+    replay(mockedObjects);
+
+    // When
+    Map<String, List<DetailedEventBookingDTO>> result = eventBookingPersistenceManager.adminGetBookingsByEventIds(eventIds);
+
+    // Then
+    DetailedEventBookingDTO expectedBooking1 = prepareDetailedEventBookingDTO(1L, now);
+    DetailedEventBookingDTO expectedBooking2 = prepareDetailedEventBookingDTO(2L, now);
+    assertEquals(2, result.size());
+    assertDeepEquals(List.of(expectedBooking1), result.get("event1"));
+    assertDeepEquals(List.of(expectedBooking2), result.get("event2"));
+
+    verify(mockedObjects);
+  }
+
+  @NotNull
+  private static DetailedEventBookingDTO prepareDetailedEventBookingDTO(long bookingId, Instant now) {
+    DetailedEventBookingDTO expectedBooking1 = new DetailedEventBookingDTO();
+    expectedBooking1.setBookingId(bookingId);
+    expectedBooking1.setUserBooked(new UserSummaryWithEmailAddressAndGenderDTO());
+    expectedBooking1.setReservedById(7L);
+    expectedBooking1.setBookingStatus(BookingStatus.CONFIRMED);
+    expectedBooking1.setBookingDate(now);
+    expectedBooking1.setUpdated(now);
+    return expectedBooking1;
+  }
+
+  private static void prepareEventBookingResultSet(ResultSet dummyResultSet, long t, String event1, Instant now)
```

Comment:
These parameters need to be renamed:
- `t`: what is t?
- `event1` to simply `event`
- `now` to something like `createAndUpdateTime`. Because this function does not know that the value of `now` is indeed `Instant.now()`. From this function's point of view, `now` is just a time that will be set to the created and updated timestamps.